crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs

```diff
@@ -267,10 +267,11 @@ pub trait SeriesJoin: SeriesSealed + Sized {
         other: &Series,
         validate: JoinValidation,
         join_nulls: bool,
+        maintain_order: bool,
     ) -> PolarsResult<(PrimitiveArray<IdxSize>, PrimitiveArray<IdxSize>)> {
```

Comment:
I don't think we have to enter this code. We can do an argsort higher up the callstack depending on the `maintain_order` state.